src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -268,21 +268,22 @@ bool dcd_edpt_open (uint8_t rhport, tusb_desc_endpoint_t const * desc_edpt)
 
   uint8_t const epnum = tu_edpt_number(desc_edpt->bEndpointAddress);
   uint8_t const dir   = tu_edpt_dir(desc_edpt->bEndpointAddress);
-
-  // Unsupported endpoint numbers/size.
-  if((desc_edpt->wMaxPacketSize.size > 64) || (epnum > EP_MAX)) {
-    return false;
-  }
-
+  
+  TU_ASSERT(desc_edpt->wMaxPacketSize.size <= 64);
+  TU_ASSERT(epnum < EP_MAX);
+  
```

Comment:
I don't care for this. Considering that `dcd_edpt_open` returns a bool, I assume there's a reason in the future that the upper levels of the stack would use that information (the current behavior _will_ `TU_BREAKPOINT` eventually).

Reply:
I've been using TU_ASSERT and TU_VERIFY based on if the condition would ever be valid behavior of the MCU's firmware. My reading of the code is that if these conditions happen, it's a bug in tinyusb or the user of tinyusb. Up to now, there's no dynamic allocation of endpoints, so this should be caught the first time the program runs in the debugger.

Are you saying that these are conditions that could happen in normal code flow?

I imagine returning false from the function implies that (for example) there was a USB protocol error or the host disconnected/reset, or something like that. Perhaps the PHY/MAC can report HW errors related to the particular EP which should cause the particular EP to stall. But, as you said, these are not handled at the moment.

Since I believe these conditions can only happen if there is a bug in the MCU firmware (or cosmic rays), it should create a breakpoint (at least in DEBUG mode), so the bug can be fixed. Release mode currently will evaluate the condition and return false (without creating a breakpoint) if the condition fails.

Reply:
>Are you saying that these are conditions that could happen in normal code flow (such as if the host sends a bad packet)?

No, the only time I had this happen was when diagnosing #95, which was a logic error.

>Release mode currently will evaluate the condition and return false (without creating a breakpoint) if the condition fails.

Oh I wasn't aware of this. That makes me feel better, though I admit I never really could use `TU_ASSERT` and `TU_VERIFY` properly lol.
Oh, then that's fine then.

Reply:
@cr1901 Though on ARM M0, the breakpoint macro does nothing...
I'll try to work up a PR which updates the documentation to describe when to use which.  I may also create a TU_DEBUG_ASSERT() which does disappear....
Another related issue I just thought of is that coverity-scan probably  doesn't realize that TU_ASSERT is assert(). I should have been asking it to treat TU_BREAKPOINT as a function which should never be called.... maybe it'll find more issues that way.

Reply:
@pigrew I dont really like the idea of DEBUG_ASSERT which expand to nothing in release mode. All the condition is dynamic, we can't say for sure we can omit it in release mode. Let's just leave it as it is. I don't want to follow the std assert.h path